.github/workflows/chromatic.yml

```diff
@@ -4,33 +4,42 @@ on:
   push:
     branches:
       - "master"
-  pull_request:
+  pull_request_target:
     types: [synchronize, labeled, unlabeled]
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
   cancel-in-progress: true
 
 jobs:
+  authorize:
+    environment:
+      ${{ github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository &&
+      'external' || 'internal' }}
+    runs-on: ubuntu-22.04
+    steps:
+      - run: echo "true"
+
   files-changed:
     name: Check which files changed
+    needs: [authorize]
     runs-on: ubuntu-22.04
     timeout-minutes: 3
     outputs:
       frontend_all: ${{ steps.changes.outputs.frontend_all }}
-      e2e_specs: ${{ steps.changes.outputs.e2e_specs }}
     steps:
       - uses: actions/checkout@v3
       - name: Test which files changed
         uses: dorny/paths-filter@v2.11.1
         id: changes
         with:
-          token: ${{ github.token }}
+          token: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
```

Comment:
I don't have a way to verify that this secret works atm, but I think you already confirmed it's correct.